src/spikeinterface/core/segmentutils.py

```diff
@@ -169,6 +169,11 @@ def get_traces(self, start_frame, end_frame, channel_indices):
         if end_frame is None:
             end_frame = self.get_num_samples()
 
+        # # Ensures that we won't request invalid segment indices
+        if (start_frame >= self.get_num_samples()) or (end_frame <= start_frame):
+            # Return (0 * num_channels) array of correct dtype
+            return self.parent_segments[0].get_traces(0, 0, channel_indices)
+
```

Comment:
Why not raising an error ? start_frame should nver be >= num_samples.
Here we are hidding another and deeper bug.
Could we revert this make a test that trigger the bug and then make a concensus fix ?

Reply:
1: Without this fix, when user calls `get_traces` with out-of-range start/end frames they get an error, which is inconsistent with the other extractors' behavior (which all return an empty array)

2: valid  TracePadRecording.get_traces calls will under the hood call get_traces with start_frame==end_frame and 
raise this error as int the issue I filed

Reply:
I am not sure to be OK this kind of hidden patch for other extractor.
start_frame >= self.get_num_samples() should not be allowed in my mind.

Reply:
Fair enough but this case is not tested for in any of the other extractors currently (including Neo based I believe)